### PR TITLE
Polygon stock load

### PR DESCRIPTION
**Pull request by jmaslek**
Adding in load `--source polygon` from polygon.io

The free tier allows 5 calls/minute and it seems that the minute calls give a lotto data, up to 50,000 entries...  We already have polygon API key in the terminal.

More to come :D

I did tests

edit:  I did more

@deeleeramone here are figi and SIK codes from polygon's endpoint
```
            AAPL Codes             
┏━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━┓
┃                  ┃              ┃
┡━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━┩
│ CIK              │ 0000320193   │
├──────────────────┼──────────────┤
│ COMPOSITE FIGI   │ BBG000B9XRY4 │
├──────────────────┼──────────────┤
│ SHARE CLASS FIGI │ BBG001S5N8V8 │
├──────────────────┼──────────────┤
│ SIC CODE         │ 3571         │
└──────────────────┴──────────────┘
```

### CODE REVIEW

File: openbb_terminal/parent_classes.py [hunk 1]
```diff
@@ -438,7 +438,9 @@ def call_load(self, other_args: List[str]):
             "--source",
             action="store",
             dest="source",
-            choices=["yf", "av", "iex"] if "-i" not in other_args else ["yf"],
+            choices=["yf", "av", "iex", "polygon"]
+            if "-i" not in other_args
```
Comment (DidierRLopes) [inline openbb_terminal/parent_classes.py:442]: or `--interval`

File: openbb_terminal/stocks/stocks_helper.py [hunk 2]
```diff
@@ -17,13 +17,15 @@
...
+import requests
+
 import yfinance as yf
 from alpha_vantage.timeseries import TimeSeries
 from numpy.core.fromnumeric import transpose
 from plotly.subplots import make_subplots
 from scipy import stats
 
-from openbb_terminal.config_terminal import theme
+from openbb_terminal.config_terminal import theme, API_POLYGON_KEY
```
Comment (DidierRLopes) [inline openbb_terminal/stocks/stocks_helper.py:28]: let's load config_terminal as cfg and then use `cfg.API_POLYGON_KEY`. That's how we doing in the other places

**Review by DidierRLopes — CHANGES_REQUESTED** [review]

File: openbb_terminal/stocks/stocks_controller.py [hunk 3]
```diff
@@ -111,7 +112,9 @@ def print_help(self):
 {self.add_info}[cmds]
     quote       view the current price for a specific stock ticker
     candle      view a candle chart for a specific stock ticker
-    news        latest news of the company[/cmds] [src][News API][/src]
+    news        latest news of the company [src][News API][/src]
+    codes       FIGI, SIK and SIC codes codes[/cmds] [src][Polygon.io][/src]
```
[the remark this answers is not in the document]
Comment (jmaslek) [reply openbb_terminal/stocks/stocks_controller.py:116]: yes.

we show the news line even if there is not a valid news api key

File: openbb_terminal/stocks/stocks_helper.py (hunk 2 above)
Comment (DidierRLopes) [reply openbb_terminal/stocks/stocks_helper.py:28]: I honestly don't like implicit imports. I think having `cfg.theme` is much more visible where that comes from, and would rather us doing that

File: openbb_terminal/stocks/stocks_controller.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (DidierRLopes) [reply openbb_terminal/stocks/stocks_controller.py:116]: Does the Polygon data here can be accessed with their free tier? if yes, nvm my comment. If not, let's hide it for users that don't have a valid paid API key

[the remark this answers is not in the document]
Comment (jmaslek) [reply openbb_terminal/stocks/stocks_controller.py:116]: Yup :D afaik everything they have can be accessed through the free tier.

File: openbb_terminal/stocks/stocks_helper.py (hunk 2 above)
Comment (jmaslek) [reply openbb_terminal/stocks/stocks_helper.py:28]: just tryna use your code against you ;)

Comment (jmaslek) [reply openbb_terminal/stocks/stocks_helper.py:28]: that also imports everything and is why we initially had such slow load times

File: openbb_terminal/stocks/stocks_controller.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (DidierRLopes) [reply openbb_terminal/stocks/stocks_controller.py:116]: So nevermind what I said!

File: openbb_terminal/stocks/stocks_helper.py (hunk 2 above)
Comment (jmaslek) [reply openbb_terminal/stocks/stocks_helper.py]: that you are <3

now approve

**Review by DidierRLopes — APPROVED** [review]

**Comment by deeleeramone** [comment]
OH BABY! That’s the missing link in the stock search! Search —> codes —> symbols —> long name —> description

**Pull request merged**